test/functional/wallet_signer.py

```diff
@@ -134,5 +135,85 @@ def run_test(self):
+        mock_wallet = self.nodes[1].get_wallet_rpc("mock")
+        assert mock_wallet.getwalletinfo()['private_keys_enabled']
+
+        result = mock_wallet.importdescriptors([{
+            "desc": "wpkh([00000001/84'/1'/0']tprv8ZgxMBicQKsPd7Uf69XL1XwhmjHopUGep8GuEiJDZmbQz6o58LninorQAfcKZWARbtRtfnLcJ5MQ2AtHcQJCCRUcMRvmDUjyEmNUWwx8UbK/0/*)#rweraev0",
+            "timestamp": 0,
+            "range": [0,1],
+            "internal": False,
+            "active": True
+        },
+        {
+            "desc": "wpkh([00000001/84'/1'/0']tprv8ZgxMBicQKsPd7Uf69XL1XwhmjHopUGep8GuEiJDZmbQz6o58LninorQAfcKZWARbtRtfnLcJ5MQ2AtHcQJCCRUcMRvmDUjyEmNUWwx8UbK/1/*)#j6uzqvuh",
+            "timestamp": 0,
+            "range": [0, 0],
+            "internal": True,
+            "active": True
+        }])
+        assert_equal(result[0], {'success': True})
+        assert_equal(result[1], {'success': True})
+        assert_equal(mock_wallet.getwalletinfo()["txcount"], 1)
+        dest = self.nodes[0].getnewaddress(address_type='bech32')
+        mock_psbt = mock_wallet.walletcreatefundedpsbt([], {dest:0.5}, 0, {}, True)['psbt']
+        mock_psbt_signed = mock_wallet.walletprocesspsbt(psbt=mock_psbt, sign=True, sighashtype="ALL", bip32derivs=True)
+        mock_psbt_final = mock_wallet.finalizepsbt(mock_psbt_signed["psbt"])
+        mock_tx = mock_psbt_final["hex"]
+        assert(mock_wallet.testmempoolaccept([mock_tx])[0]["allowed"])
+
+        # # Create a new wallet and populate with specific public keys, in order
+        # # to work with the mock signed PSBT.
+        # self.nodes[1].createwallet(wallet_name="hww4", disable_private_keys=True, descriptors=True, external_signer=True)
+        # hww4 = self.nodes[1].get_wallet_rpc("hww4")
+        #
+        # descriptors = [{
+        #     "desc": "wpkh([00000001/84'/1'/0']tpubD6NzVbkrYhZ4WaWSyoBvQwbpLkojyoTZPRsgXELWz3Popb3qkjcJyJUGLnL4qHHoQvao8ESaAstxYSnhyswJ76uZPStJRJCTKvosUCJZL5B/0/*)#x30uthjs",
+        #     "timestamp": "now",
+        #     "range": [0, 1],
+        #     "internal": False,
+        #     "watchonly": True,
+        #     "active": True
+        # },
+        # {
+        #     "desc": "wpkh([00000001/84'/1'/0']tpubD6NzVbkrYhZ4WaWSyoBvQwbpLkojyoTZPRsgXELWz3Popb3qkjcJyJUGLnL4qHHoQvao8ESaAstxYSnhyswJ76uZPStJRJCTKvosUCJZL5B/1/*)#h92akzzg",
+        #     "timestamp": "now",
+        #     "range": [0, 0],
+        #     "internal": True,
+        #     "watchonly": True,
+        #     "active": True
+        # }]
+
+        # result = hww4.importdescriptors(descriptors)
+        # assert_equal(result[0], {'success': True})
+        # assert_equal(result[1], {'success': True})
+        assert_equal(hww.getwalletinfo()["txcount"], 1)
+
+        assert(hww.testmempoolaccept([mock_tx])[0]["allowed"])
+
+        f = open(os.path.join(self.nodes[1].cwd, "mock_psbt"), "w")
+        f.write(mock_psbt_signed["psbt"])
+        f.close()
```

Comment:
In 57b26f44b9109eb0c366adeb54bf5ebfc816b45e "rpc: send: support external signer"

Use `with`

```suggestion
        with open(os.path.join(os.getcwd(), "mock_psbt"), "w") as f:
             f.write(mock_psbt_signed["psbt"])
```